### DISCUSSION TITLE
Calendar component doesn't work as expected in tests

### DISCUSSION THREAD

**Discussion by iamyegor** [primefaces/community#456] [PrimeReact] [answered]
I'm encountering an issue while testing a React component that uses the PrimeReact Calendar component. The calendar is meant to allow users to pick dates, and it works as expected in the actual browser. However, when running tests using React Testing Library and Vitest, the dropdown part of the calendar (the actual date picker) is not being rendered in the DOM.

Here's a brief overview of the component:
```
import { Calendar } from "primereact/calendar";
import { useState } from "react";
import { useSearchParams } from "react-router-dom";

export default function BookCalendars() {
  // [State and handlers setup]

  return (
    <div>
        <p>Trip start date</p>
        <Calendar
          data-testid="start-date"
          value={startDate}
          minDate={new Date()}
          onChange={(e) => handleStartDateChange(e)}
          dateFormat="mm/dd/yy"
        />
    </div>
  );
}
```
During testing, the rendered output for the calendar is as follows:
```
<span class="p-calendar p-component p-inputwrapper p-inputwrapper-filled">
  <input class="p-inputtext p-component" type="text" />
</span>
```

As you can see, the input field of the calendar is present, but the dropdown portion (where dates are actually picked) does not appear in the test's DOM.

I've tried simulating user interactions (like clicking the input field) to trigger the dropdown, but it still doesn't render. The issue seems to be specific to the testing environment since the component works correctly in the actual browser.

**Comment by melloware** [comment]
It won't appear in the DOM unless you use `inline` mode or click the input to focus the calendar which then renders the popup panel.  Basically the panel is not rendered until it is needed.

**Reply by iamyegor** [reply]
I click on the calendar in my tests but it still doesn't show up. It only works when I use inline mode, but I don't like it.

**Comment by iamyegor** [comment, marked as answer]
I fixed the issue by adding a label like in this example:
```
<span className="p-float-label">
    <Calendar inputId="birth_date" value={date} onChange={(e) => setDate(e.value)} />
    <label htmlFor="birth_date">Birth Date</label>
</span>
```
Now I just click on the label in my test and the popup renders.
